toke.c

```diff
@@ -2012,10 +2012,11 @@ S_update_debugger_info(pTHX_ SV *orig_sv, const char *const buf, STRLEN len)
     AV *av = CopFILEAVx(PL_curcop);
     if (av) {
         SV * sv;
-        if (PL_parser->preambling == NOLINE) sv = newSV_type(SVt_PVMG);
+        if (PL_parser->preambling == NOLINE)
+            sv = newSV_type(SVt_PVIV);
         else {
             sv = *av_fetch(av, 0, 1);
-            SvUPGRADE(sv, SVt_PVMG);
+            SvUPGRADE(sv, SVt_PVIV);
         }
         if (!SvPOK(sv)) SvPVCLEAR(sv);
```

Comment:
Unroll `SvUPGRADE()`, add `{}` to `SvPVCLEAR(sv);`, add 2 `goto` skipping `if(!SvPOK(sv))` test and jump right into the branch, if was `< SVt_PV` or we did a `newSV_type(SVt_PVIV);`. We know the SV* is undef and/or doesn't have `POK_on`, so we can skip the `if(!SvPOK(sv))` conditional jump CPU opcode/opcodes.